pkg/dockerfile/dockerfile.go

```diff
@@ -87,6 +105,38 @@ func Stages(opts *config.KanikoOptions) ([]config.KanikoStage, error) {
 	return kanikoStages, nil
 }
 
+func lintDockerfile(dockerfilePath string, ignoredRules []string) ([]byte, error) {
+	args := []string{dockerfilePath}
+	if len(ignoredRules) > 0 {
+		type LintConfig struct {
+			IgnoredRules []string `yaml:"ignored"`
+		}
+		var lintCfg LintConfig
+		lintCfg.IgnoredRules = ignoredRules
+
+		data, err := yaml.Marshal(&lintCfg)
+		if err != nil {
+			return nil, errors.Wrap(err, fmt.Sprintf("marshaling hadolint config %v", lintCfg))
+		}
+
+		cfgPath := "/kaniko/.hadolint.yaml"
```

Comment:
what if the user provides .hadolint.yaml config through a volume mount e.g. `docker run -v $HOME/.hadolint.yaml:/kaniko/.hadolint.yaml kaniko-executor ...`? It seems that might be a possible scenario which should be catched by checking if the file exist before creating it and use the already existing file instead.

Reply:
Thank you very much for this review.
I will just explain here the reason why I didn't take into account this kind of scenario : I found that trusted registry from hadolint doesn't work perfectly with mentionned bug on description of this PR and then I've decided to exclude it from kaniko to avoid kaniko users to complain about it.
Adding for users the possibility to have their own hadolint config means they can add trusted registries inside. 
Now after your review I'm just in mode "OK let's do this anyway, if they complain it will be hadolint issue and not kaniko's one".
I will add the possibility for client to override hadolint config, but in same time it means I have to add trustedRegstry multiple args in kaniko cli.
Will work on it starting tomorrow.

Reply:
Let's wait for @tejal29 thoughts as well before you start doing a major refactor since I'm a new contributor and she might have totally different view on this. I didn't originally consider your note about trustedRegistry when I reviewed the PR, that's a good point. My understanding is trustedRegistry setting won't work with GCR but will work with other registry settings, which is irrelevant of kaniko, so it would still be valuable for users to be able to get full flexibility to provide a `.hadolint.yaml`?

Reply:
I think you are right. We shouldn't block users from using their own hadolint config, even if one of hadolint feature isn't totally working right now, because as you said, irrelevant of kaniko